pyomo/core/plugins/transform/bounds_tightener.py

```diff
@@ -0,0 +1,61 @@
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+
+import textwrap
+
+class BoundsTightener(IsomorphicTransformation):
+    """Tightens bound on constraint, by iterating through each variable and
+    tightening the constraint bounds accordingly"""
+
+
+    alias('core.bounds_tightener',doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+    def __init__(self):
+        super(BoundsTightener, self).__init__()
+
+    def _apply_to(self,instance):
+
+        for constr in instance.component_data_objects(ctype = Constraint,
+                                                        active = True,
+                                                        descend_into=True):
+            if not constr.body.polynomial_degree() == 1:
+                continue
+
+            #tighten the constraint bound as much as possible
+            repn = generate_canonical_repn(constr.body)
+            UB = 0
+            if repn.constant:
+                UB = -1 * repn.constant
+            LB = UB
+            for i, coef in enumerate(repn.linear):
+                #TODO: handle unbounded vars
+                #TODO: ROunding issues
+
+                if coef >= 0:
+                    if repn.variables[i].ub and repn.variables[i].lb:
```

Comment:
Why do you only update the UB/LB if the variable has both?  You should be able to tighten one side even if the other side is still unbounded.